lib/Controller/LoginController.php

```diff
@@ -366,7 +366,11 @@ private function login($uid, Profile $profile, $newGroupPrefix = '')
                 } catch (\Exception $e) {}
             }
 
-            if (!empty($profile->data['groups']) && is_array($profile->data['groups'])) {
+            # we assume here that $profile->data['groups'] can either be:
+            # - null, when there is no information on groups the user is a member of
+            # - empty array, which means the user explicitly is not a member of any groups
+            # - a non-empty array, listing the groups the user is a member of
+            if (is_array($profile->data['groups'])) {
```

Comment:
Bad idea. You can get many "Undefined index 'groups'" in log.
Fuck, I got. `empty([])` returns `false`. I'll fix it